src/sentry/services/hybrid_cloud/__init__.py

```diff
@@ -135,19 +139,50 @@ def service_stubbed(
         raise ValueError("Service needs to be a DelegatedBySilMode object, but it was not!")
 
 
-@contextlib.contextmanager
-def use_real_service(
-    service: InterfaceWithLifecycle, silo_mode: SiloMode
-) -> Generator[None, None, None]:
-    """
-    Removes any stubbed implementations, forcing the default configured implementation.
-    Important for integration tests that validate the integration of production service implementations.
-    """
-    from django.test import override_settings
-
-    if isinstance(service, DelegatedBySiloMode):
-        with override_settings(SILO_MODE=silo_mode):
-            with service.with_replacement(None, silo_mode):
-                yield
-    else:
-        raise ValueError("Service needs to be a DelegatedBySiloMode object, but it was not!")
+class use_real_service:
```

Comment:
This change simply allows this to be used bother as a test decorator and as a context.